shared/providers.ts

```diff
@@ -301,21 +321,22 @@ export function createHoverProvider(
         ): AsyncGenerator<sourcegraph.Badged<sourcegraph.Hover> | null | undefined, void, undefined> {
             const emitter = new TelemetryEmitter()
 
-            let hasPreciseResult = false
-            for await (const lsifResult of lsifProvider(textDocument, position)) {
-                if (lsifResult) {
-                    await emitter.emitOnce('lsifHover')
-                    if (hasPreciseResult) {
-                        yield lsifResult
-                    } else {
-                        yield { ...lsifResult, alerts: HoverAlerts.lsif }
+            const lsifWrapper = await lsifProvider(textDocument, position)
+            if (lsifWrapper) {
```

Comment:
My understanding is that LSIF functionality replaces LSP - is it a common use case where we have both LSP and LSIF providers for a language on one instance? If we don't expect that to be true then this logic could just be an either/or; where first check for optional LSP provider results in true, skip check for LSIF. If no LSP then try LSIF.